### PR TITLE
[Doc][Tune][Restore] Split "Tune stopping and resuming" into two user guides

### LINKED ISSUE #30223: [Tune] Scope of `Tuner.restore()` functionality is unclear in docs
### What happened + What you expected to happen

The resumed run has identical best values as the first run. In another word, the tuning doesn't make further progress when resuming the runs.  I expect every time the code is run, it will resume from the last checkpoint and produce better results. 

### Versions / Dependencies

ray2.10

### Reproduction script

      import os
      from ray import tune, air
      from hyperopt import hp
      from ray.tune.search.hyperopt import HyperOptSearch
      from ray.air import session
      from ray.air.checkpoint import Checkpoint

      # 1. Define an objective function.
      def objective(config):
          score = config["a"] ** 2 + config["b"]
          session.report({'SCORE':score})

      # 2. Define a search space.
      search_space = {
          "a": hp.uniform("a", 0, 1),
          "b": hp.uniform("b", 0, 1)
          }

      raw_log_dir = "ray_log"
      raw_log_name = "example"
      log_dir = os.path.join(os.getcwd(), raw_log_dir, raw_log_name)
      if os.path.exists(log_dir) == False:
          print('--- this is the 1st run ----')
          algorithm = HyperOptSearch(search_space, metric="SCORE", mode="max")
          tuner = tune.Tuner(objective,
                  tune_config = tune.TuneConfig(
                      num_samples = 1, # number of tries. too expensive for Brian2
                      search_alg=algorithm,
                      ),
                  param_space=search_space,
                  run_config = air.RunConfig(local_dir = raw_log_dir, name = raw_log_name) # where to save the log which will be loaded later
                  )
      else: #note: restoring described here doesn't work: https://docs.ray.io/en/latest/tune/tutorials/tune-stopping.html
          print('--- previous run exist, continue the tuning ----')
          tuner = tune.Tuner.restore(log_dir)

      results = tuner.fit()
      print(results.get_best_result(metric="SCORE", mode="max").config)

### Issue Severity

High: It blocks me from completing my task.

### LINKED ISSUE #31830: [AIR] AIR fault tolerance needs a centralized user guide
### Description

I can't find much information about specifying failure config for worker-level fault tolerance. It's briefly described in one user guide under Ray Train here: https://docs.ray.io/en/latest/train/config_guide.html#failure-configuration-failureconfig.

I want a user guide (under AIR or Tune) structured like:

# Fault Tolerance in Ray AIR/Tune

## Experiment-level Fault Tolerance

`Tuner.restore`

What fault tolerance guarantees can users assume?

## Trial-level Fault Tolerance

`FailureConfig` and trial checkpointing.

What fault tolerance guarantees can users assume?

## Example

An example with both enabled using Ray Jobs + retry.

### Link

_No response_

### PR DESCRIPTION
**Pull request by justinvyu**
## Summary

This PR splits the "Tune Stopping and Resuming" guide into two user guides: (1) Tune Stopping, and (2) Tune Fault Tolerance. This PR also adds a FAQ post about the "iterative experimentation" workflow not being supported by `Tuner.restore`, but rather through *new* Tune experiments.

### Why?

The scope of `Tuner.restore` functionality is unclear. Users think that it should be used for iterative experimentation where hyperparameters/configs are changed before resuming training. However, its usage should be limited to providing fault tolerance at the experiment level, without any changes to the experiment specification. [[github](https://github.com/ray-project/ray/issues/30223), [slack](https://ray-distributed.slack.com/archives/CNECXMW22/p1667813395820229), [slack](https://ray-distributed.slack.com/archives/CMVUQ22JD/p1657540799493319), [slack](https://ray-distributed.slack.com/archives/C01DLHZHRBJ/p1672007501253379)]

In previous discussions ([Tuner API Design](https://docs.google.com/document/d/1UvVydw7SwiPQy_knDlzl65BtEvIwqPipZEFLFKqQFAs/edit#heading=h.j6bn6j4m7so5)), we have decided against allowing modifications to things like trial stopping criteria upon resume. The idea is that an experiment should follow a single spec, and experiment restore is used for fault-tolerance to continue retrying the job if it’s interrupted. We have not communicated this decision through to outward-facing docs and usage examples (both in the open-source and product documentation), which is the reason for the continued user questions (see links above).

### --> For reviewers <--

There are 3 new pieces of the docs to review:

- [New Tune stopping guide](https://anyscale-ray--33495.com.readthedocs.build/en/33495/tune/tutorials/tune-stopping.html)
- [New Tune fault tolerance guide](https://anyscale-ray--33495.com.readthedocs.build/en/33495/tune/tutorials/tune-fault-tolerance.html)
- [New FAQ section](https://anyscale-ray--33495.com.readthedocs.build/en/33495/tune/faq.html#how-can-i-continue-training-a-completed-tune-experiment-for-longer-and-with-new-configurations-iterative-experimentation)

Here's what the old guide looks like: https://docs.ray.io/en/releases-2.3.0/tune/tutorials/tune-stopping.html

## Related issue number

Closes https://github.com/ray-project/ray/issues/30223
Closes https://github.com/ray-project/ray/issues/31830

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — COMMENTED** [review]
very nice man.
so super quick questions

File: doc/source/tune/doc_code/faq.py [hunk 1]
```diff
@@ -439,3 +439,62 @@ def train_func(config):
...
+        )
+
+
+new_tuner = tune.Tuner(
+    trainable,
+    param_space={
+        "num_epochs": 10,
+        "hyperparam": tune.grid_search([4, 5, 6]),
+        # Use the `best_checkpoint_uri` from above.
+        "start_from_checkpoint": best_checkpoint_path,
```
[the remark this answers is not in the document]
Comment (justinvyu) [reply doc/source/tune/doc_code/faq.py:495]: Nope, this is just some random config name for users to seed the new run with a checkpoint. This actually only works if the checkpoint is available in s3. Otherwise, if it's a local path, some nodes will not find it.

File: doc/source/tune/doc_code/fault_tolerance.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    tuner = tune.Tuner(
+        trainable,
+        param_space={"num_epochs": 10},
+        run_config=air.RunConfig(
+            local_dir="~/ray_results", name="single_script_fit_or_restore"
+        ),
+    )
+tuner.fit()
+# __ft_restore_multiplexing_start__
+
```
[the remark this answers is not in the document]
Comment (justinvyu) [reply doc/source/tune/doc_code/fault_tolerance.py:73]: Got it, can do.

[the remark this answers is not in the document]
Comment (justinvyu) [reply doc/source/tune/doc_code/fault_tolerance.py:73]: For re-specifying datasets, I'd have to use a Trainer. Ideally this would go somewhere in AIR instead?

File: doc/source/tune/faq.rst [hunk 3]
```diff
@@ -805,3 +805,33 @@ leading to unexpected results.
...
+Now, I want to continue training from a checkpoint (e.g., the best one) generated by the previous experiment,
+and search over a new hyperparameter search space, for another ``10`` epochs.
+
+:ref:`tune-fault-tolerance` explains that the usage of :meth:`Tuner.restore <ray.tune.Tuner.restore>`
+is meant for resuming an *unfinished* experiment that was interrupted in the middle,
+according to the *exact configuration* that was supplied in the initial training run.
+
+Therefore, ``Tuner.restore`` is not suitable for our desired behavior.
+This style of "iterative experimentation" should be done with *new* Tune experiments
+rather than restoring a single experiment over and over and modifying the experiment spec.
```
[the remark this answers is not in the document]
Comment (justinvyu) [reply doc/source/tune/faq.rst:830]: See the first point. Without introducing some API to pass an initial checkpoint to Tuner, then accessing it through `session.get_checkpoint`, we need to just pass it through the config.

**Review by richardliaw — APPROVED** [review]
